event-handlers/src/test/java/no/sikt/nva/nvi/events/cristin/CristinNviReportEventConsumerTest.java

```diff
@@ -0,0 +1,118 @@
+package no.sikt.nva.nvi.events.cristin;
+
+import static java.util.UUID.randomUUID;
+import static no.sikt.nva.nvi.events.cristin.CristinMapper.AFFILIATION_DELIMITER;
+import static no.sikt.nva.nvi.events.cristin.CristinMapper.API_HOST;
+import static no.sikt.nva.nvi.test.TestUtils.randomYear;
+import static no.unit.nva.testutils.RandomDataGenerator.objectMapper;
+import static no.unit.nva.testutils.RandomDataGenerator.randomInstant;
+import static no.unit.nva.testutils.RandomDataGenerator.randomString;
+import static nva.commons.core.attempt.Try.attempt;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.mockito.Mockito.mock;
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.events.SQSEvent;
+import com.amazonaws.services.lambda.runtime.events.SQSEvent.SQSMessage;
+import java.net.URI;
+import java.time.LocalDate;
+import java.util.List;
+import no.sikt.nva.nvi.common.db.CandidateRepository;
+import no.sikt.nva.nvi.common.db.PeriodRepository;
+import no.sikt.nva.nvi.common.service.model.Candidate;
+import no.sikt.nva.nvi.test.LocalDynamoTest;
+import nva.commons.core.paths.UriWrapper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class CristinNviReportEventConsumerTest extends LocalDynamoTest {
+
+    public static final LocalDate DATE_CONTROLLED = LocalDate.now();
+    public static final String STATUS_CONTROLLED = "J";
+    private static final Context CONTEXT = mock(Context.class);
+    public static final String PUBLICATION = "publication";
+    private CristinNviReportEventConsumer handler;
+    private CandidateRepository candidateRepository;
+    private PeriodRepository periodRepository;
+
+    @BeforeEach
+    void setup() {
+        localDynamo = initializeTestDatabase();
+        candidateRepository = new CandidateRepository(localDynamo);
+        periodRepository = new PeriodRepository(localDynamo);
+        handler = new CristinNviReportEventConsumer(candidateRepository);
+    }
+
+    @Test
+    void shouldCreateNviCandidateFromNviReport() {
+        var cristinNviReport = randomCristinNviReport();
+        handler.handleRequest(createEvent(cristinNviReport), CONTEXT);
+        var publicationId = toPublicationId(cristinNviReport);
+        var nviCandidate = Candidate.fetchByPublicationId(() -> publicationId, candidateRepository, periodRepository);
+
+        assertThatNviCandidateHasExpectedValues(nviCandidate, cristinNviReport);
```

Comment:
Assert that all fields are as expected? `applicable`, `publicationBucketUri`, `level`, `instanceType`, `publicationDate`? I would create an expectedCandidate to assert everything is expected